src/inverted-index.js

```diff
@@ -72,31 +80,80 @@ class InvertedIndex {
       });
       // Sort the index keys
       index = InvertedIndex.sortObjectKeys(index);
-      // Simply update the this.indexes object
-      // Don't return it yet
+      /** Simply update the this.indexes object
+       * Don't return it yet
+       */
       this.indexes[filename] = index;
     }
   }
 
   /**
+   * Returns all indexes or an index for a specified file
+   *
+   * @param{String} filename - Optional parameter specifying a file whose index is required
    * @returns{Object} - Object that contains all the index
    */
-  getIndex() {
-    return this.indexes;
+  getIndex(filename) {
+    if (filename === undefined) {
+      return this.indexes;
+    }
+    return this.indexes[filename];
+  }
+
+  /**
+   * Function to iterate over the indexes object and returns the search results
+   *
+   * @param{Array} words - An array of words to be searched
+   * @param{Object} indexes - An object containing inverted indexes
+   * @param{String} fileName - Optional argument indicating the filename
+   * @returns{Object} results - An object containing the search results
+   */
+  static iterCollection(words, indexes, fileName) {
+    // Object to store search results
+    const results = {};
+    // check if fileName is undefined
+    if (fileName === undefined) {
+      // search all indexes
+      for (const file in indexes) {
+        results[file] = {};
+        words.forEach((word) => {
+          if (Object.keys(indexes[file]).includes(word)) {
+            results[file][word] = indexes[file][word];
+          } else {
+            results[file][word] = 'Word not found!';
+          }
+        });
+      }
+    } else {
+      // get the index for the file specified
+      const fileIndex = indexes[fileName];
+      words.forEach((word) => {
```

Comment:
### Critical

Your `iterCollection` still does not solve the duplication issue. Look closely how lines `119 - 125` & `130 - 136` are exactly the same thing.